### ISSUE TITLE
Add ignore option ["at-rule", "comment", "rule"] to indentation

### ISSUE THREAD

**Issue by davidtheclark** [stylelint/stylelint#280]
So that users with idiosyncratic statement indentation patterns can still have the rest of their indentation checked.

cf. https://github.com/stylelint/stylelint/issues/278

**Comment by jeddy3** [comment]
I'm now wondering whether we should make this a little more specific than my original suggestion of `statement` e.g.

`ignore:  ["comment', "rule"]`

Allowing a user to ignore things like:

```css
.foo { }

    /* comment */
    .bar {}
```

Plus, it'll be a simple `node.type` look-up for us.

Whatcha think?

**Comment by davidtheclark** [comment]
I don't see why not.

**Comment by davidtheclark** [comment]
I imagine that the idea is that given 

```css
a {
  color: pink;
}

  @media print {
    a {
      color: orange;
    }
  }
```

the rule would still enforce the indentation of that second `a` rule. It would have to base the expected indentation on the parent, while remaining ignorant of whether the parent is properly indented or not (in this case, the parent itself doesn't care).

How, then, would we catch errors like this:

```css
  @media print {
  a {
    color: pink;
  }
}
```

Let's say that indentation on the first line is a mistake. But the `a` rule doesn't know that, so it would expect to be indented with two indents, and it would complain, too, when in fact it is correct.

Does this make sense?

**Comment by davidtheclark** [comment]
I'll try to summarize a little:

I was thinking that if we want to enforce a block's children while ignoring the block's own indentation level, then our calculation of how many indents a line requires would need to be *relative to the parent* rather than *relative to the root* (as it currently is). But this means that if the parent is *wrong*, then *the children are expected to be wrong accordingly.* I don't think that's great, because it means that the children are being tagged with erroneous warnings.

That said, I've only have the time to spend about 20 minutes thinking about this problem. Maybe one of you have an idea? Or maybe I'll think of something better later.

**Comment by MoOx** [comment]
> It would have to base the expected indentation on the parent

That's the hard part. In the case you explained I can't see how to do...

I think we should go for a specified parameter in order to know expectations.
If we add a rule to require indentation for similar selectors (see https://github.com/stylelint/stylelint/issues/278#issuecomment-121669073) we might know that `a` should be 6 spaces indented (based on the number of parents, so we might also detect the previous node indentation.

**Comment by jeddy3** [comment]
>  would need to be relative to the parent rather than relative to the root (as it currently is)

My bad. I misunderstood the inner workings of `indentation` when I suggested this "ignore" approach.

Now I'm thinking that we keep `indentation` as is and those with idiosyncratic indentation practices can avoid the rule. Unless you want to scratch this itch that is.

**Comment by davidtheclark** [comment]
Agreed. So we're closing this, and plan to work on a pluggable rule (https://github.com/stylelint/stylelint/issues/295) for the special style @MoOx uses.
